### PR TITLE
Privacy

### PR DESCRIPTION
**Pull request by AdamLeonSmith**

### CODE REVIEW

**Review by ShivanKaul — COMMENTED** [review]
Is there a reason we need a separate file per privacy principle? I think it might be simpler to navigate if we had all of them in the same document.

File: principles/3_Privacy.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Privacy 
+
+## Data Minimization
+
+* The minimum amount of data possible shall be collected, for the purposes for which it is gathered. Data that is not used in the disclosure to healthcare authorities should not be collected unless it is essential to the purposes of the application
+* Users can choose how much data to share from symptom surveys
+
+## Consent
+* A member of the public (MoP) shall consent to data being collected or transmitted explicitly, that is it should not be implied by usage, or mixed with another consent
+* A MoP shall provide informed consent, that is they must be able to understand what they are consenting to and the purposes for which each category of data is gathered.
```
Comment (lmartensson) [inline principles/3_Privacy.md]: Either use period in all bullet points or none.

File: principles/3_Privacy.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Privacy 
+
+## Data Minimization
+
+* The minimum amount of data possible shall be collected, for the purposes for which it is gathered. Data that is not used in the disclosure to healthcare authorities should not be collected unless it is essential to the purposes of the application
+* Users can choose how much data to share from symptom surveys
+
+## Consent
+* A member of the public (MoP) shall consent to data being collected or transmitted explicitly, that is it should not be implied by usage, or mixed with another consent
```
Comment (lmartensson) [inline principles/3_Privacy.md]: Suggested edit for clarity: "A member of the public (MoP) shall be explicitly asked for consent to data being collected or transmitted. Consent should not be implied by usage, or mixed with any other consent requests."

Comment (AdamLeonSmith) [reply principles/3_Privacy.md]: done

File: principles/3_Privacy.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+* A MoP shall provide informed consent, that is they must be able to understand what they are consenting to and the purposes for which each category of data is gathered
+* Information about privacy shall be communicated in plain English, not legal, or technical language
```
Comment (lmartensson) [inline principles/3_Privacy.md]: Suggested edit for clarity:
"Information about privacy shall be communicated in as simple language as possible while maintaining clarity. Specifically, legal or technical language should be avoided."

I guess my gripe with "plain English" is that that could still include fancy words that are not commonly known, also that English might not be the relevant language.

File: principles/3_Privacy.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+* Consent to collect data shall be separate from consent to disclose data, and include retention periods
```
Comment (lmartensson) [inline principles/3_Privacy.md:13]: ", and include retention periods" to be a separate bullet point?

File: principles/3_Privacy.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+* Consent to collect each category of data (e.g. location, proximity) shall be separate
+* No communication initiated by 3rd parties shall be provided to a MoP, through the app, without their consent
+* No data shall be shared with 3rd parties, except healthcare authorities using the agreed mechanisms
+* The user of SafePaths shall be able to disable location tracking temporarily
```
Comment (lmartensson) [inline principles/3_Privacy.md:17]: Also delete data?

File: principles/3_Privacy.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+* Consent to disclose data to a HA shall be separate, at the time of disclosure, and contain a reference to the HA privacy policy
+
+## Transparency & Accountability
+
+* The privacy policy shall include the steps taken by the data collector to ensure the confidentiality, integrity, and quality of the data
+* The data controller, that is, who controls what data is collected and how it is stored and disclosed shall be identified clearly
```
Comment (lmartensson) [inline principles/3_Privacy.md]: Suggested edit for clarity:
"It shall be identified clearly who controls what data is collected and how it is stored and disclosed."

Comment (lmartensson) [reply principles/3_Privacy.md]: My readability nit here is mainly that "the data controller" and "identified clearly" are on opposite sides of the sentence.

File: principles/3_Privacy.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+* No single person shall be able to modify the source code and release the application to the public
```
Comment (lmartensson) [inline principles/3_Privacy.md]: Should accountability be a separate section?

Comment (lmartensson) [reply principles/3_Privacy.md]: Also: "to **both** modify the source and release"?

In addition, code should be reviewed before being checked in?

File: principles/3_Privacy.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+* All source code, models and technical documentation shall be publicly available, including all dependencies
```
Comment (lmartensson) [inline principles/3_Privacy.md:26]: I'm not sure about "including all dependencies". That could unnecessarily restrict what libraries and infrastructure we can use? Especially on the server side, e.g. if we want to use various GCP or AWS technologies they may not be open source. I could also see having different policies for the client side and the server side here.

File: principles/3_Privacy.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+* An assessment of any residual privacy risks shall be made public in the form of a risk assessment
+
+# Anonymization
+
+* Location, proximity, health, race/ethnicity, gender, and any data that allows the identification of a person, shall be considered sensitive unless it is sufficiently aggregated
+* Paths between waypoints shall not be stored or published
+* No identifiers of nearby devices shall be recorded that can be used to identify a MoP
+* Healthcare authorities should have the ability to effectively redact data
+* Un-redacted data shall not be persisted by a Healthcare authority in the Safe Places web application or it’s supporting components
```
Comment (lmartensson) [inline principles/3_Privacy.md:35]: I'm not sure what this means. All data is deleted after x days? Also, what about future analysis of data? I don't know to what extent we intend to make that possible, but it sounds like this could prevent that.

File: principles/3_Privacy.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+* Location or proximity data of less than N number of diagnosed people shall not able to be published, where N can be configured based on local legislation
```
Comment (lmartensson) [inline principles/3_Privacy.md:36]: I'm not sure what this means either. That actual location data could potentially be published as long as there are enough diagnosed people? That could very easily identify individual infected people even in aggregation. "Look, Bob lives alone in the woods miles from any other people, and the data shows his house had an infection."

File: principles/3_Privacy.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+* Precautions shall be taken to ensure that shared aggregate data may not be re-identified downstream, by ensuring that no data items are passed from SafePaths to SafePlaces that are not required (see DM2)
+* Users should have the ability to effectively redact data themselves
```
Comment (lmartensson) [inline principles/3_Privacy.md:38]: Fits better in the consent section?

Comment (lmartensson) [reply principles/3_Privacy.md:38]: Also, does this mean we intend to provide a function for them to do this in the app, as opposed to "transmit a request to healthcare authorities to retract the data" as stated in the next section?

File: principles/3_Privacy.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+* Retention periods shall be 14 days in the mobile application
+* Retention periods shall be in the privacy policy
+* Users shall be able to transmit a request to healthcare authorities to retract the data
+* Healthcare authorities have the technical ability retract data they have published
+* SafePlaces shall support a technical process for users to sign-off on post-redacted data.
+* Retention periods in SafePlaces shall be configurable and enforced technically
+
+# Other Technical Measures
+* Data shall be as accurate as possible
+* Privacy shall be as inclusive as possible, and this includes making the application work just as well for people with different abilities, wherever technically possible, and based on W3C/WAI standards
```
Comment (lmartensson) [inline principles/3_Privacy.md]: Include link to these standards?

File: principles/3_Privacy.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+* Measures shall be taken to avoid algorithms leading to unfair outcomes for people based on immutable characteristics of people (e.g. race, gender, age) or socio-economic class This includes proxy variables such as location, which may infer these items
```
Comment (lmartensson) [inline principles/3_Privacy.md:54]: Missed period after "class", and end of line.

**Review by lmartensson — COMMENTED** [review]
Good doc! I have a couple of questions in there but mostly suggestions based on clarity.

File: principles/3_Privacy.md (hunk 6 above)
Comment (skheinze) [reply principles/3_Privacy.md]: Perhaps "The privacy policy shall identify clearly both the person who controls the data that is collected and how the data is stored and disclosed. "

File: principles/3_Privacy.md (hunk 9 above)
Comment (skheinze) [reply principles/3_Privacy.md:35]: When I read this before, I assumed that 32 and 33 referred to individual data and not to aggregated data, and that individual data is deleted after x days while aggregated data is stored longterm. If this is correct, we could specify which we are referring to for clarity.

File: principles/3_Privacy.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+* Healthcare authorities should have the ability to effectively redact data
+* Un-redacted data shall not be persisted by a Healthcare authority in the Safe Places web application or it’s supporting components
+* Location or proximity data of less than N number of diagnosed people shall not able to be published, where N can be configured based on local legislation
+* Precautions shall be taken to ensure that shared aggregate data may not be re-identified downstream, by ensuring that no data items are passed from SafePaths to SafePlaces that are not required (see DM2)
+* Users should have the ability to effectively redact data themselves
+
+# Retention & Retraction
+
+* Retention periods shall be 14 days in the mobile application
+* Retention periods shall be in the privacy policy
```
Comment (skheinze) [inline principles/3_Privacy.md]: "retention periods shall be identified in the privacy policy"

File: principles/3_Privacy.md (hunk 3 above)
Comment (altoplano) [reply principles/3_Privacy.md]: Also not all users are english speakers. The "plain language" term sounds better to me.
Small concern that "plain english" may have some legal weight that "plain language" doesn't?

File: principles/3_Privacy.md (hunk 5 above)
Comment (altoplano) [reply principles/3_Privacy.md:17]: Would be reat to allow MoP to selectively delete parts of their data before uploading it to the HA but the UI/UX for something that works on most phones, in multiple cultures, with varying levels of technical sophistication seems to much to tackle. Instead I think we are depending on them trusting the CT interviewer to do the deletion they ask for.

File: principles/3_Privacy.md (hunk 8 above)
Comment (altoplano) [reply principles/3_Privacy.md:26]: Would being explicit about what services from GCP or AWS were being used meet the "be publicly available" need?
Seems this is to assure that the MoP can evaluate what they are entrusting their data to.
Those services are public offerings, so if well identified would that serve this purpose?

File: principles/3_Privacy.md (hunk 9 above)
Comment (altoplano) [reply principles/3_Privacy.md:35]: A MoP might be asked, seperately from SafePaths/SafePlaces if they consent to their data being kept for some other explicit purpose like long term analysis? It is their data after all.
HA's have all kinds of regulations around the handling of health data, they may well need to redact even the aggregated data at times, like if they are asked to remove someones data from the aggregation, or if they determine some error has occurred.

Comment (altoplano) [reply principles/3_Privacy.md:35]: Regarding #30 - Seems like this precludes storing information about public transit routes. Do we want to do that (if it does)?

File: principles/3_Privacy.md (hunk 10 above)
Comment (altoplano) [reply principles/3_Privacy.md:36]: #32 has some implications on the data flow of raw path data from the phone to CT interview. Perhaps it is encrypted with a key the user has to provide to the CT interviewer and SafePlaces can not store that key making the data nominally "noise".

**Review by altoplano — APPROVED** [review]
Looks like still some outstanding comments but I'm fine with accepting this as an early draft.

File: principles/3_Privacy.md (hunk 3 above)
Comment (AdamLeonSmith) [reply principles/3_Privacy.md]: Very good points, I have amended as proposed.  I don't think plain english has any legal standing.

File: principles/3_Privacy.md (hunk 5 above)
Comment (AdamLeonSmith) [reply principles/3_Privacy.md:17]: Under Anonymization we actually have "Users should have the ability to effectively redact data themselves" - does that cover it?  Not achievable for MVP 1 though.

File: principles/3_Privacy.md (hunk 6 above)
Comment (AdamLeonSmith) [reply principles/3_Privacy.md]: I've updated in line with @skheinze proposal which I think is slightly clearer!

File: principles/3_Privacy.md (hunk 7 above)
Comment (AdamLeonSmith) [reply principles/3_Privacy.md]: I don't think we should split the section as e.g. the privacy policy speaks to transparency and legal enforceability.

Proposed wording change implemented.

File: principles/3_Privacy.md (hunk 8 above)
Comment (AdamLeonSmith) [reply principles/3_Privacy.md:26]: I like the current "all dependencies".  Let's say various AWS ML services started to be used, that would be very non-transparent and not in line with our mission.  If we used something not privacy impacting, e.g. S3 services, then we could risk assess that appropriately and list it as a deviation.  thoughts?

File: principles/3_Privacy.md (hunk 9 above)
Comment (AdamLeonSmith) [reply principles/3_Privacy.md:35]: > Regarding #30 - Seems like this precludes storing information about public transit routes. Do we want to do that (if it does)?

Amended to " Paths followed by individuals between waypoints shall not be stored or published"

Comment (AdamLeonSmith) [reply principles/3_Privacy.md:35]: > When I read this before, I assumed that 32 and 33 referred to individual data and not to aggregated data, and that individual data is deleted after x days while aggregated data is stored longterm. If this is correct, we could specify which we are referring to for clarity.

I meant both - updated to be clearer

Comment (AdamLeonSmith) [reply principles/3_Privacy.md:35]: > A MoP might be asked, seperately from SafePaths/SafePlaces if they consent to their data being kept for some other explicit purpose like long term analysis? It is their data after all.

I suppose that could happen, but it isn't supported by our technical solution.  I can imagine that if that consent was gained outside of our process, the HA could build a mechanism to extract data...

File: principles/3_Privacy.md (hunk 10 above)
Comment (AdamLeonSmith) [reply principles/3_Privacy.md:36]: > I'm not sure what this means either. That actual location data could potentially be published as long as there are enough diagnosed people? That could very easily identify individual infected people even in aggregation. "Look, Bob lives alone in the woods miles from any other people, and the data shows his house had an infection."

Yes that's true.  So we have multiple levels of control (a) based on local legal views they can only publish data once we have a certain number of infected patients; and (b) the data is not in plain text - it has been through a one way hash.

Comment (AdamLeonSmith) [reply principles/3_Privacy.md:36]: > #32 has some implications on the data flow of raw path data from the phone to CT interview. Perhaps it is encrypted with a key the user has to provide to the CT interviewer and SafePlaces can not store that key making the data nominally "noise".

Do you think we should add that as a new principle?  i.e. that data should be encrypted on the device with a key that the user gives in the verbal consent mechanism?  I believe MVP1 takes steps towards that.

File: principles/3_Privacy.md (hunk 11 above)
Comment (AdamLeonSmith) [reply principles/3_Privacy.md:38]: There is a plan to provide the ability for the user to redact the data themselves, and to provide consent on the redacted data itself.  Moved that last bit to consent as you suggest!

File: principles/3_Privacy.md (hunk 8 above)
Comment (skheinze) [reply principles/3_Privacy.md:26]: I think it is safer to use "all dependencies" and force the addition of a deviation than to list all current specific services. I think that has a better chance of causing someone to think of what they are doing as a deviation from the standard.

**Review by skheinze — APPROVED** [review]
While there are still some open comments, I think we can proceed at this point.

File: principles/3_Privacy.md (hunk 10 above)
Comment (altoplano) [reply principles/3_Privacy.md:36]: I think this would be a way to implement the privacy and security of the MoP path data but there may be other, more effective ways.
Having something twoards that end as a principal does seem like a good addition.

The current MVP1 concept, as I understand it, is that the CT interviewer has to provide a token that allows the HA a way to prevent their input channel from being spammed with unwanted input. It explicitly does not use it to encrypt the data currently.
